MMTabBarView/MMTabBarView/MMTabDragAssistant.m

```diff
@@ -85,110 +85,121 @@ - (NSDragOperation)draggingSession:(NSDraggingSession *)session sourceOperationM
 
 	if (_draggedTab) {
-		[[_draggedTab window] setFrameTopLeftPoint:screenPoint];
-		[[_draggedTab window] orderFront:nil];
+		[_draggedTab.window setFrameTopLeftPoint:screenPoint];
+		[_draggedTab.window orderFront:nil];
 
-		if ([[tabBarView tabView] numberOfTabViewItems] == 1) {
-			[self _draggingExitedTabBarView:tabBarView withPasteboardItem:[self pasteboardItem]];
+		if (tabBarView.tabView.numberOfTabViewItems == 1) {
+			MMTabPasteboardItem* const pasteboardItem = self.pasteboardItem;
+			if (pasteboardItem != nil) {
+				[self _draggingExitedTabBarView:tabBarView withPasteboardItem:pasteboardItem];
+			}
 		}
 	}
 }
 
 - (void)draggingSession:(NSDraggingSession *)session movedToPoint:(NSPoint)screenPoint {
 	if (_draggedTab) {
 		if (_centersDragWindows) {
-			if ([_draggedTab isAnimating]) {
+			if (_draggedTab.isAnimating) {
 				return;
 			}
 			//Ignore aPoint, as it seems to give wacky values
-			NSRect frame = [[_draggedTab window] frame];
+			NSRect frame = _draggedTab.window.frame;
 			frame.origin = screenPoint; //[NSEvent mouseLocation];
 			frame.origin.x -= frame.size.width / 2;
 			frame.origin.y -= frame.size.height / 2;
-			[[_draggedTab window] setFrame:frame display:NO];
+			[_draggedTab.window setFrame:frame display:NO];
 		} else {
-			[[_draggedTab window] setFrameTopLeftPoint:screenPoint];
+			[_draggedTab.window setFrameTopLeftPoint:screenPoint];
 		}
 
 		if (_draggedView) {
 			//move the view representation with the tab
 			//the relative position of the dragged view window will be different
 			//depending on the position of the tab bar relative to the controlled tab view
-			screenPoint.y -= [[_draggedTab window] frame].size.height;
+			screenPoint.y -= _draggedTab.window.frame.size.height;
 			screenPoint.x -= _dragWindowOffset.width;
 			screenPoint.y += _dragWindowOffset.height;
-			[[_draggedView window] setFrameTopLeftPoint:screenPoint];
+			[_draggedView.window setFrameTopLeftPoint:screenPoint];
 		}
 	}
 }
 
 - (void)draggingSession:(NSDraggingSession *)session endedAtPoint:(NSPoint)screenPoint operation:(NSDragOperation)operation {
 
-    MMTabPasteboardItem *pasteboardItem = [self pasteboardItem];
+    MMTabPasteboardItem *pasteboardItem = self.pasteboardItem;
     
-	NSTabView *sourceTabView = [_sourceTabBar tabView];
-	NSUInteger sourceIndex = [pasteboardItem sourceIndex];
+	NSTabView *sourceTabView = _sourceTabBar.tabView;
+	NSUInteger sourceIndex = pasteboardItem.sourceIndex;
 
-    if ([self isDragging]) {   // means there was not a successful drop (performDragOperation)
+    if (self.isDragging) {   // means there was not a successful drop (performDragOperation)
     
-        id <MMTabBarViewDelegate> sourceDelegate = [_sourceTabBar delegate];
+        id <MMTabBarViewDelegate> sourceDelegate = _sourceTabBar.delegate;
 
         //split off the dragged tab into a new window
```

Comment:
Modified logic here

Reply:
OK for me

Reply:
This one is identical, just using dot syntax, isn´t it?

Reply:
Yes the above is identical. But my little concern was with the following lines. OK for me now.